TypeCobol/Compiler/Preprocessor/CopyNameMapFile.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Diagnostics;
+using TypeCobol.Compiler.Directives;
+using TypeCobol.Tools.Options_Config;
+
+namespace TypeCobol.Compiler.Preprocessor
+{
+    /// <summary>
+    /// Class that implements, mapping of CPY copy names.
+    /// The Lookup is performed using  BinarySearch, thus it necessary for the underlying array of names to be sorted.
+    /// ALL COPY NAMES MUST BE IN UPPERCASE.
+    /// </summary>
+    public class CopyNameMapFile
+    {
+        private string[] _CpyCopyNames;        
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="filepath">The Copy list File Path to map</param>
+        /// <param name="bSort">true to sort the resulting array of Copy names, false otherwise.</param>
+        public CopyNameMapFile(String filepath, bool bSort = false)
+        {
+            _CpyCopyNames = System.IO.File.ReadAllLines(filepath);
+            if (_CpyCopyNames != null && bSort)
+                Array.Sort(_CpyCopyNames);
+        }
+
+        /// <summary>
+        /// Array constructor
+        /// </summary>
+        /// <param name="cpyCopyNames">The Array of Cpy Names</param>
+        /// <param name="bSort">true to sort the array</param>
+        public CopyNameMapFile(string[] cpyCopyNames, bool bSort = false)
+        {
+            _CpyCopyNames = cpyCopyNames;
+            if (_CpyCopyNames != null && bSort)
+                Array.Sort(_CpyCopyNames);
+        }
+
+        /// <summary>
+        /// Get the singleton instance. Create one if it does not exists
+        /// </summary>
+        /// <returns>The singleton instance</returns>
+        public static CopyNameMapFile Singleton { get; set; }
```

Comment:
But in this case this is exactly a Singleton there is only one Cpy List file for CLI or for the Server, and it simplify tests creation.

Reply:
I agree with @smedilol, this would change the current option model. Besides, the CompilerDirectiveBuilders already have an instance of `TypeCobolOptions`.

Reply:
List of intrinsic, list of dependencies, list of analyzers, ... are not singleton even if you can only set one list per CLI.